src/auro-datetime.js

```diff
@@ -76,97 +64,78 @@ class AuroDatetime extends LitElement {
+      newDate = new Date(this.utc);
 
-  /**
-   * @private Internal function generate numeric date string
-   * @returns {string} - date string
-   */
-  numericDate() {
-    this.dateTemplate.month = 'numeric';
-    Reflect.deleteProperty(this.dateTemplate, 'weekday');
-    const newDate = new Date();
+    } else if (this.setDate) {
+      newDate = new Date(this.setDate);
+
+    }
 
     return newDate.toLocaleString('en-us', this.dateTemplate);
   }
 
   /**
-   * @private Internal function generate month date string
+   * @private Internal function to determine new Date object based on input type
    * @returns {string} - date string
    */
-  humanDateMonth() {
-    this.template.month = this.month;
-    const newDate = new Date();
-
-    return newDate.toLocaleString('en-us', this.template);
-  }
+  humanDateConversion() {
+    let newDate = new Date();
 
-  /**
-   * @private Internal function generate year date string
-   * @returns {string} - date string
-   */
-  humanDateYear() {
-    this.template.year = 'numeric';
-    const newDate = new Date();
+    if (this.utc) {
+      this.template.timeZone = 'UTC';
+      newDate = new Date(this.utc);
+    } else if (this.setDate) {
+      newDate = new Date(this.setDate);
+    }
 
-    return newDate.toLocaleString('en-us', this.template);
-  }
+    switch (this.type) {
+      case 'day':
+        this.template.day = "numeric";
+        break;
+      case 'month':
+        this.template.month = this.month;
+        break;
+      case 'year':
+        this.template.year = 'numeric';
+        break;
+      case 'weekday':
+        this.template.weekday = this.weekday;
+        break;
 
-  /**
-   * @private Internal function generate weekday date string
-   * @returns {string} - date string
-   */
-  humanDateWeekday() {
-    this.template.weekday = this.weekday;
-    const newDate = new Date();
+      default: this.template.weekday = this.template;
+    }
 
     return newDate.toLocaleString('en-us', this.template);
   }
 
   /**
-   * @private Internal function generate day date string
+   * @private Internal function generate numeric date string 00/00/0000
    * @returns {string} - date string
    */
-  humanDateDay() {
-    this.template.day = "numeric";
-    const newDate = new Date();
-
-    return newDate.toLocaleString('en-us', this.template);
-  }
+  numericDate() {
+    this.dateTemplate.month = 'numeric';
+    Reflect.deleteProperty(this.dateTemplate, 'weekday');
+    let newDate = new Date();
 
-  /**
-   * @private Internal function generate standard date string from input data
-   * @param {string} dateData - Input date data
-   * @returns {string} - date string
-   */
-  setHumanDate(dateData) {
-    const newDate = new Date(dateData);
+    if (this.utc) {
+      this.dateTemplate.timeZone = 'UTC';
+      newDate = new Date(this.utc);
 
-    return newDate.toLocaleString('en-us', this.dateTemplate);
-  }
+      return newDate.toLocaleString('en-us', this.dateTemplate);
```

Comment:
One more place where we can use the return at the end of the method instead of returning in the `if` block.